kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/FloatingPointParser.kt

```diff
@@ -74,7 +169,10 @@ object FloatingPointParser {
      * StringExponentPair will be used to calculate the floating point number by
      * taking the positive integer the String represents and multiplying by 10
      * raised to the power of the exponent.
-
+     *
+     * This method is not optimal, can cause performance issues. This logic can be done in one single run.
```

Comment:
@ilmat192 I would also like to refactor this method in the future, as I can see that it can lead to performance degradation in the parsing of strings. Do you have any benchmarks or environments where I can check my fixes?

I do not want to break anything in this PR, so I decided not to refactor this part of the complex logic, but I would definitely like to.

Reply:
Hello!

> Do you have any benchmarks  or environments where I can check my fixes?

Unfortunetely, we don't have benchmarks. connected with parsing string with doubles now. 
But we have the benchmarks set where it's possible to add. It can be added in https://github.com/JetBrains/kotlin/blob/master/kotlin-native/performance/ring/src/main/kotlin/org/jetbrains/ring/StringBenchmark.kt or as separate class. We can add them or you can do it if you want to check particular cases.  (It's also needed to call method here https://github.com/JetBrains/kotlin/blob/master/kotlin-native/performance/ring/src/main/kotlin/main.kt)

The detailed info about running benchmarks can be found https://github.com/JetBrains/kotlin/blob/master/kotlin-native/HACKING.md#performance-measurement.

Reply:
BTW: I would like refactor this method in the next PR (I am afraid to break existing functionality in this bugfix and do not want to make huge PRs). 

What is the process you have for refactoring?
Do I need to submit an issue on youtrack for refactoring or just can simply create PR?

Reply:
I think that just creating a PR is ok.